### PR TITLE
Replace the legacy pointer input event handling with PointerPackets

### PR DESCRIPTION
**Pull request by jason-simmons**
With this change we no longer need engine/core/events and its IDL bindings.

### CODE REVIEW

File: sky/engine/core/window/window.cc [hunk 1]
```diff
@@ -67,17 +66,42 @@ void Window::UpdateWindowMetrics(const SkyDisplayMetrics& metrics) {
...
+  void* data;
+  intptr_t len;
+  if (Dart_IsError(Dart_TypedDataAcquireData(data_handle, &type, &data, &len)))
+    return;
+
+  packet->Serialize(data, len);
+
+  Dart_TypedDataReleaseData(data_handle);
+
+  DartInvokeField(library_.value(), "_dispatchPointerPacket", {data_handle});
```
Comment (chinmaygarde) [inline sky/engine/core/window/window.cc]: This `DartInvokeField` acts as an embedder entry point. `//sky/engine/bindings/dart_vm_entry_points.txt` needs to be updated.

**Comment by chinmaygarde** [comment]
lgtm. I really like flushing all pending pointer events in one go.

I will update the iOS platform stuff once this lands.

**Comment by abarth** [comment]
LGTM

**Comment by abarth** [comment]
You'll also need to update `//services/sky`, which is what builds the `sky_viewer.mojo` app that runs inside MojoShell.  It's still using the legacy view manager, so you'll probably need to create pointer packets out of whatever input events the legacy view manager is using.

**Comment by jamesr** [comment]
sky_viewer.mojo doesn't talk to the legacy view manager (or any other view manager), it talks to native viewport.  The input events it gets are https://github.com/domokit/mojo/blob/master/mojo/services/input_events/interfaces/input_events.mojom instances from the NativeViewportEventDispatcher interface: https://github.com/domokit/mojo/blob/master/mojo/services/native_viewport/interfaces/native_viewport.mojom#L50

**Comment by abarth** [comment]
@jamesr Yes, you're right.  At some point, we should probably teach NativeViewportEventDispatcher (and eventually Mozart) about pointer packets.

File: sky/engine/core/window/window.cc (hunk 1 above)
Comment (jason-simmons) [inline sky/engine/core/window/window.cc]: done

**Comment by jason-simmons** [comment]
A few updates:
* Removed some unused types from sky/services/engine/input_event.mojom
* Implemented usage of PointerPacket for sky_viewer.mojo

**Comment by abarth** [comment]
LGTM

**Pull request merged**